gslib/utils/copy_helper.py

```diff
@@ -3222,6 +3234,9 @@ def _ValidateAndCompleteDownload(logger,
   if file_name != final_file_name:
     # Data is still in a temporary file, so move it to a permanent location.
     os.rename(file_name, final_file_name)
+  if use_stet:
+    # Decrypt data using STET binary.
+    stet_util.decrypt_download(src_url, dst_url, logger)
```

Comment:
Can we do this in the caller? I think this breaks the contract a bit given that this method is only meant to download an object given a source and destination. WE shouldn't try to do anything more than that here.

Reply:
No, `ValidateAndCompleteDownload` is the best place to perform a transform. We already perform the "ungzip" transform and "temp file rename" transform here.

Reply:
I think gzip does make sense since it is required for integrity checking.

In case of stet, I think we are already telling `_DownloadObjectToFile` to use a source and a destination and download the file to that destination. I feel whatever outside transformation we need to do, we should be doing that after the download is finished as the DownloadObjectToFile has done the task of successfully downloading the file. Also we need to pass the `use_stet` flag two levels down which can be avoided.

As mentioned in the upload section, I am not strongly against what you have, so I am fine keeping it if other reviewers agree on this.

Reply:
BTW, what's the dst_url here? Is it the one with .gstmp extension? Are we renaming the tmp file to final_file_name already on line 3239?

Reply:
The decryption can't happen anywhere else b/c ungzip has to run first.

`final_file_name` is based on `dst_url`.

Yes, we are renaming the temporary file to the final file in the line `os.rename(file_name, final_file_name)`.

I would rather `file_name` be called `temporary_file_name` but that would require some extra refactors.